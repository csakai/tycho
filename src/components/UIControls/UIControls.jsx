import React from 'react';
import cx from 'classnames';
import Settings from '../Settings';
import ZoomSlider from '../Slider/ZoomSlider';
import DatePickerContainer from '../../containers/DatePickerContainer';
import PlayPauseContainer from '../../containers/PlayPauseContainer';
import VolumeContainer from '../../containers/VolumeContainer';
import Constants from '../../constants';

export default class UIControls extends React.Component {
  render() {
    return (
      <div className={cx({
        'uicontrols': true,
        'uicontrols--enabled': this.props.controlsEnabled,
        'uicontrols--disabled': !this.props.controlsEnabled,
      })}>
        <div className="uicontrols__control uicontrols__control--scales">
          <Settings {...this.props} />
        </div>

        <div className="uicontrols__control uicontrols__control--target-label">
          <span
            className="uicontrols__control uicontrols__control--stats-modal"
            onClick={this.props.openModal.bind(this, Constants.UI.ModalTypes.STATS_MODAL)}>
            {this.props.targetName}
          </span>
        </div>

        <div className="uicontrols__control uicontrols__control--datetime">
          <DatePickerContainer
            time={this.props.time}
            onUpdate={this.props.changeTimeOffset}
          />
          <PlayPauseContainer />
        </div>

        <div className="uicontrols__control uicontrols__control--left-bar">
          <div
            className="uicontrols__button uicontrols__button--about-modal"
            onClick={this.props.openModal.bind(this, Constants.UI.ModalTypes.ABOUT_MODAL)}>
          </div>
          <ZoomSlider
            value={this.props.zoom}
            onChange={this.props.changeZoom}
          />
          <div className="uicontrols__button uicontrols__button--volume">
            <VolumeContainer />
          </div>
        </div>
      </div>
    );
  }
}
