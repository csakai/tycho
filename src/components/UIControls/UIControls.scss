@import 'globals.scss';

.uicontrols {
  opacity: 0;
  transition: opacity $transition-speed;
  z-index: $z-index-top;

  &--enabled {
    opacity: 1;
  }

  &--disabled {
    opacity: 0;
  }

  &__button {
    background-color: $brand-primary;
    width: $icon-size;
    height: $icon-size;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: $base-spacing;
    cursor: pointer;

    &--about-modal {
      color: $brand-secondary;
      font-weight: bold;
      border-radius: 50%;
      font-style: italic;

      &:before {
        content: 'i';
      }

      &:hover {
        box-shadow: $text-shadow;
      }
    }

    &--volume {
        /* one day this will be useful... */
        height: 0;
        margin: 0;
        background-color: transparent;
    }
  }

  &__control {
    position: absolute;
    opacity: 0.5;
    transition: opacity $transition-speed;

    &:hover {
      opacity: 1;
    }

    &--datetime {
      display: flex;
      align-items: center;
      top: 0;
      right: 0;
    }

    &--target-label {
      bottom: 0;
      left: 0;
      right: 0;
      padding: 3em 0;
      text-align: center;
      pointer-events: none;
    }

    &--stats-modal {
      opacity: 1;
      pointer-events: auto;
      font-size: $font-size-medium;
      transition: text-shadow $transition-speed;
      color: $brand-primary;
      cursor: pointer;

      &:hover {
        text-shadow: $text-shadow;
      }
    }

    &--scales {
      bottom: 0;
      right: 0;
      pointer-events: none;
    }

    &--left-bar {
      left: 0;
      top: 0;
      bottom: 0;
      height: 100%;
      display: flex;
      flex-direction: column;
    }
  }
}
