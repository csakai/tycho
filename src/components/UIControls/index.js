import UIControls from './UIControls';
import './UIControls.css';

export default UIControls;
