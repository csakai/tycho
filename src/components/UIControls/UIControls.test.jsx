import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import UIControls from './UIControls';

describe('UIControls Component', () => {
    describe('render()', () => {
        it('should render the UIControls successfully', () => {
            const component = shallow(<UIControls
                pageText={{}}
                openModal={jest.fn()}
            />);
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
