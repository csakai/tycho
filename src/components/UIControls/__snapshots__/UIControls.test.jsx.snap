// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIControls Component render() should render the UIControls successfully 1`] = `
<div
  className="uicontrols uicontrols--disabled"
>
  <div
    className="uicontrols__control uicontrols__control--scales"
  >
    <Settings
      openModal={[Function]}
      pageText={Object {}}
    />
  </div>
  <div
    className="uicontrols__control uicontrols__control--target-label"
  >
    <span
      className="uicontrols__control uicontrols__control--stats-modal"
      onClick={[Function]}
    />
  </div>
  <div
    className="uicontrols__control uicontrols__control--datetime"
  >
    <DatePickerContainer />
    <Connect(PlayPauseContainer) />
  </div>
  <div
    className="uicontrols__control uicontrols__control--left-bar"
  >
    <div
      className="uicontrols__button uicontrols__button--about-modal"
      onClick={[Function]}
    />
    <ZoomSlider />
    <div
      className="uicontrols__button"
    />
  </div>
</div>
`;
